### PR TITLE
Add visualization function and add structure to the synthetic dataset

### PR DESCRIPTION
**Pull request by ahwillia**
**TL;DR** For full example usage see `examples/demo_synthetic.ipynb`.

**New functions for the CMF model:** `model.plot(...)` and `model.argsort_units()`

Here is an example call to `model.plot(data, tmin=0, tmax=...)`. The left plots are the fitted W factors and (three components here) and the top plots are the fitted H factors.

image

I also added a sorting option which lexographically sorts the rows of the data matrix based on (first) which component the neuron has the largest loading on, and (second) the peak activation time over the lag dimension of W. After sorting the rows we get:

image

cc @bantin @bsorsch @degleris1

### CODE REVIEW

File: cmfpy/model.py [hunk 1]
```diff
@@ -236,6 +243,36 @@ def factors(self):
         except AttributeError:
             raise NOT_FITTED_ERROR
 
+    def argsort_units(self):
```
Comment (degleris1) [inline cmfpy/model.py]: This looks good and is more concise than the `sort_neurons` function in `visual.py`. I'll set a note to remove the old function at some point.

File: cmfpy/visualize.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+"""
+Functions to visualize model results.
+"""
```
Comment (degleris1) [inline cmfpy/visualize.py]: Do we want to put this in `utils/visual.py`?
